src/components/ManaCalculator/calculator.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  GENERATION_PER_SLOT,
+  first_slot_of_epoch,
+  potential_Mana,
+  targetReward,
+  EPOCH_DURATION,
+  decay,
+} from './utils';
+import { ValidatorParameters, ValidatorProps } from './types';
+
+export function calculateManaRewards(
+  stake: number,
+  yourPool: number,
+  validatorParameters: ValidatorParameters,
+  validators: ValidatorProps[],
+  initialEpoch: number,
+  finalEpoch: number,
+  yourRole: 'Validator' | 'Delegator',
+  supply: number,
+): number {
+  let totalTargetReward = 0;
+  const epochDiff = 1;
```

Comment:
is this really supposed to be `1`?

Reply:
I have no clue why I did it like that 😆 Should be the actually diff. Let me fix that 😅

Reply:
Fixed in b47899d1c55ea4d6dc8f564106ad2a26cc8148b8